parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -0,0 +1,544 @@
+    )
+    argparser.add_argument(
+        '--correctness_is_flipped',
+        default=False,
+        action='store_true',
+        help='question phrasing flips the better model - e.g. question is "Which speaker is more boring?"',
+    )
+    argparser.add_argument(
+        '--model_comparisons',
+        type=str,
+        help='list of model pairs to compare, comma separated. E.g. ["transformer,human_eval"] ',
+    )
+    argparser.add_argument(
+        '--pairs_per_matchup',
+        type=int,
+        default=160,
+        help='Number of conversation pairs to generate for the comparison',
+    )
+    argparser.add_argument(
+        '--num_onboarding_tasks',
+        type=int,
+        default=5,
+        help='Number of onboarding tasks total to screen workers with',
+    )
+    argparser.add_argument(
+        '--block_on_onboarding_fail',
+        type=bool,
+        default=True,
+        help='whether to block on onboarding failure',
+    )
+    argparser.add_argument(
+        '--onboarding_tasks',
+        type=list,
+        default=None,
+        help='onboarding tasks to screen workers with, list of (conv1id, conv2id, matchup) tuples',
+    )
+    argparser.add_argument(
+        '--onboarding_model_comparison',
+        type=str,
+        default='greedy_model,human_eval',
+        help='models to compare for the onboarding task. E.g. "greedy,human_eval" ',
+    )
+    argparser.add_argument(
+        '--subtasks_per_hit',
+        type=int,
+        default=5,
+        help='number of subtasks/comparisons to do per hit',
+    )
+    argparser.add_argument(
+        '--onboarding_threshold',
+        type=float,
+        default=0.75,
+        help='minimum accuracy on onboarding tasks, as a float 0-1.0',
+    )
+    argparser.add_argument('--seed', type=int, default=42, help='np.random seed')
+    argparser.set_defaults(allowed_conversation=1)
+    if from_argv:
+        return argparser.parse_args()
+    else:
+        return argparser.parse_args(args=[])
+
+
+def list_files(folder):
+    for fn in os.listdir(folder):
+        full_fn = os.path.join(folder, fn)
+        if os.path.isfile(full_fn):
+            yield full_fn
+        elif os.path.isdir(full_fn):
+            for sfn in list_files(full_fn):
+                yield sfn
+
+
+def setup_task_queue(opt):
+    """ Initialize task queue to contain the specified number of instances of
+    each pairing
+    """
+    # hacky fix for the parlai parser hacky fix
+    data_folder = opt['dialogs_path'].replace('-', '_')
+    annotations_per_pair = opt['annotations_per_pair']
+    all_conv_data = {}
+    conv_ids_by_model = {}
+    internal_id = 0
+
+    # read in all conversation data
+    for data_fn in os.listdir(data_folder):
+        full_data_fn = os.path.join(data_folder, data_fn)
+        with open(full_data_fn, 'r') as dialog_data_file:
+            model_name = data_fn.split('.')[0]
```

Comment:
so the data is expected to be in the data folder with specific file formatting? can we document this somewhere?

Reply:
yes, I need to write a good README

Reply:
what if instead we made this *not* dependent on the datafile name? shouldn't the model name be specified in the json anyway? can't we just provide some root dir to search through for json's with that model name?

Reply:
I would be ok with instead specifying a dict with model_name: file_name - is that sort of what you had in mind?

Reply:
I was thinking more along the lines of: each JSON conversation has the key `model_name`, right? Can we just specify the root directory, search all conversations, and select by `model_name` key?

Reply:
It would be nice to have default functionality that does the right thing if you give it a directory and also if you give it a file. In addition, one nice thing would be if someone just has the ability to implement a single function that parses their file. (So pull this out into its own function that can be overridden.) I have to parse a CSV file someone else gave me for example. The frontend expects certain fields to be there such as "dialog",  so the format should be specified too as per emily's comment above.